Server/cron/cronFunctions/customMap.ts

```diff
@@ -3,8 +3,31 @@ import { discordClient } from "../../discord";
 import { MappoolMap } from "../../../Models/tournaments/mappools/mappoolMap";
 import { Tournament } from "../../../Models/tournaments/tournament";
 import { mappoolLog } from "../../../DiscordBot/functions/tournamentFunctions";
+import { CronJobData, CronJobType } from "../../../Interfaces/cron";
 
-export default async function execute () {
+async function initialize (): Promise<CronJobData[]> {
+    // Get all maps with a deadline
+    const maps: { deadline: Date }[] = await MappoolMap
+        .createQueryBuilder("map")
+        .select("distinct deadline")
+        .where("map.deadline IS NOT NULL")
+        .getRawMany();
+
+    // For each map, create a cron job with the deadline as the date.
+    let cronJobs: CronJobData[] = maps.map(map => ({
+        type: CronJobType.Custommap,
+        date: map.deadline!,
+    })).filter((j, i, a) => a.findIndex(j2 => j2.date.getTime() === j.date.getTime()) === i);
+
+    if (cronJobs.some(j => j.date.getTime() < Date.now())) {
+        await execute();
```

Comment:
this is out-of-scope, i think it would be cleaner to add another cronjob to the array, scheduled to be executed instantly.